File: maven-confluence-core/src/test/java/org/bsc/maven/plugin/confluence/Issue99Test.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package org.bsc.maven.plugin.confluence;

import org.bsc.confluence.ConfluenceHtmlListUtils;
import java.util.regex.*;
import org.hamcrest.core.IsEqual;
import org.junit.Assert;
import org.junit.Ignore;
import org.junit.Test;

/**
 *
 * @author bsorrentino
 */
public class Issue99Test {

    @Ignore @Test
    public void dummy() {}

    public void tokenizeUsingRegex() {
        String line = "\"hello world\" Alexandros Alex \"I Am\" Something";
        Pattern pattern = Pattern.compile("\\b(?:(?<=\")[^\"]*(?=\")|\\w+)\\b");
        Matcher matcher = pattern.matcher(line);

        int i = 0;
        while (matcher.find()) {
            System.out.println(matcher.group(0));
            ++i;
        }

        Assert.assertThat( i, IsEqual.equalTo(5));
    }

    @Test
    public void tokenizeHtmlUsingRegex() {
        final String line = new StringBuilder("this is an example of comment")
                .append( "<ol>").append('\n')
                .append("<li>item1</li>").append('\n')
                .append("<li>item2</li>").append('\n')
                .append("<li>item3</li>").append('\n')
                .append("</ol>").append('\n')
                .append( "another example")
                .append( "<ul>").append('\n')
                .append("<li>item1</li>").append('\n')
                .append("<li>item2</li>").append('\n')
                .append("<li>item3</li>").append('\n')
                .append("</ul>").append('\n')
                .append( "end example")
                .toString()
                ;

                final String result = ConfluenceHtmlListUtils.replaceHtmlList(line);

                final String expect = new StringBuilder()
                        .append("this is an example of comment")
                        .append('\n')
                        .append( "# item1" ).append('\n')
                        .append( "# item2" ).append('\n')
                        .append( "# item3" ).append('\n')
                        .append('\n')
                        .append( "another example")
                        .append('\n')
                        .append( "* item1" ).append('\n')
                        .append( "* item2" ).append('\n')
                        .append( "* item3" ).append('\n')
                        .append('\n')
                        .append( "end example")
                        .toString()
                        ;
                Assert.assertThat( result, IsEqual.equalTo(expect));


    }

    @Test
    public void tokenizeHtmlUsingRegex1() {
        final String line = new StringBuilder()
                .append( "<ul>" ).append('\n')
                .append("<li><code>&#42;&#42;/&#42;.?ar</code></li>" ).append('\n')
                .append( "<li><code>&#42;&#42;/&#42;.dll</code></li>" ).append('\n')
                .append( "</ul>" ).append('\n')
                .toString()
                ;

        final String result = ConfluenceHtmlListUtils.replaceHtmlList(line);

        final String expect = new StringBuilder()
                .append('\n')
                .append( "* <code>&#42;&#42;/&#42;.?ar</code>" ).append('\n')
                .append( "* <code>&#42;&#42;/&#42;.dll</code>" ).append('\n')
                .append('\n')
                .toString()
                ;
        Assert.assertThat( result, IsEqual.equalTo(expect));


    }

    @Test
    public void tokenizeHtmlUsingRegex2() {
        final String line = new StringBuilder()
                .append("<b>this is an example of bold</b>" ).append('\n')
                .toString()
                ;

        //final String result = line.replaceAll("<[Bb]>|</[Bb]>", "*");
        final String result = line.replaceAll("</?[Bb]>", "*");

        final String expect = new StringBuilder()
                .append( "*this is an example of bold*" ).append('\n')
                .toString()
                ;
        Assert.assertThat( result, IsEqual.equalTo(expect));


    }

    @Test
    public void tokenizeHtmlUsingRegex3() {
        final String input = "this is an example of comment";
        final String line = new StringBuilder(input)
                .toString()
                ;

        final String result =  ConfluenceHtmlListUtils.replaceHtmlList(line);

        Assert.assertThat( result , IsEqual.equalTo(input));

    }

}

File: core/src/test/kotlin/maven/plugin/confluence/Issue99Test.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package maven.plugin.confluence

import org.bsc.confluence.ConfluenceHtmlUtils.replaceHtmlList
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Disabled
import org.junit.jupiter.api.Test
import java.util.regex.Pattern

/**
 *
 * @author bsorrentino
 */
class Issue99Test {
    @Disabled
    @Test
    fun dummy() {
    }

    fun tokenizeUsingRegex() {
        val line = "\"hello world\" Alexandros Alex \"I Am\" Something"
        val pattern = Pattern.compile("\\b(?:(?<=\")[^\"]*(?=\")|\\w+)\\b")
        val matcher = pattern.matcher(line)
        var i = 0
        while (matcher.find()) {
            println(matcher.group(0))
            ++i
        }
        assertEquals(5, i)
    }

    @Test
    fun tokenizeHtmlUsingRegex() {
        val line = """
            this is an example of comment
            <ol>
            <li>item1</li>
            <li>item2</li>
            <li>item3</li>
            </ol>
            another example
            <ul>
            <li>item1</li>
            <li>item2</li>
            <li>item3</li>
            </ul>
            end example
        """.trimIndent()

        val expect = """
                this is an example of comment
                
                # item1
                # item2
                # item3
                
                another example
                
                * item1
                * item2
                * item3
                
                end example
            """.trimIndent()
        assertEquals( expect, replaceHtmlList(line) )
    }

    @Test
    fun tokenizeHtmlUsingRegex1() {
        val line = """
            <ul>
            <li><code>&#42;&#42;/&#42;.?ar</code></li>
            <li><code>&#42;&#42;/&#42;.dll</code></li>
            </ul>
        """.trimIndent()

        val expect = """
            
            * <code>&#42;&#42;/&#42;.?ar</code>
            * <code>&#42;&#42;/&#42;.dll</code>
            
        """.trimIndent()
        assertEquals( expect, replaceHtmlList(line) )
    }

    @Test
    fun tokenizeHtmlUsingRegex2() {
        val line = """
            <b>this is an example of bold</b>
        """.trimIndent()

        //final String result = line.replaceAll("<[Bb]>|</[Bb]>", "*");
        val result = line.replace("</?[Bb]>".toRegex(), "*")

        val expect = """
            *this is an example of bold*
        """.trimIndent()
        assertEquals( expect, result )
    }

    @Test
    fun tokenizeHtmlUsingRegex3() {
        val input = "this is an example of comment"
        assertEquals(input, replaceHtmlList(input))
    }
}